src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Parser/ParserHelpers.cs

```diff
@@ -26,7 +26,7 @@ internal static class ParserHelpers
         public const int Int64OverflowLength = 19;
         public const int Int64OverflowLengthHex = 16;
 
-        public static readonly byte[] s_hexLookup =
+        public static ReadOnlySpan<byte> HexLookup => new byte[] // rely on C# compiler optimization to reference static data
```

Comment:
Out of curiosity, why this instead of the following?

```suggestion
        public static ReadOnlySpan<byte> HexLookup { get; } = new byte[] // rely on C# compiler optimization to reference static data
```

Reply:
Because that prevents the optimization: https://github.com/dotnet/roslyn/pull/24621.

Reply:
More generally, you can't actually do that with a span.

Reply:
> More generally, you can't actually do that with a span. It won't compile.

Ah, right, thanks! Was getting it confused with `ReadOnlyMemory<byte>`.